src/generic/inplace-text-editor/index.tsx

```diff
@@ -51,41 +80,91 @@ export const InplaceTextEditor: React.FC<InplaceTextEditorProps> = ({
     }
   };
 
+  if (readOnly) {
+    return (
+      <span className={textClassName}>
+        {text}
+      </span>
+    );
+  }
+
+  if (alwaysShowEditButton) {
+    return (
+      <Stack
+        direction="horizontal"
+        gap={1}
+      >
+        {inputIsActive
+          ? (
+            <Form.Control
+              autoFocus
+              type="text"
+              aria-label="Text input"
+              defaultValue={text}
+              onBlur={handleOnChangeText}
+              onKeyDown={handleOnKeyDown}
+            />
+          )
+          : (
+            <span className={textClassName}>
+              {text}
+            </span>
+          )}
+        <IconButton
+          src={Edit}
+          iconAs={Icon}
+          alt={intl.formatMessage(messages.editTextButtonAlt)}
+          onClick={handleEdit}
+          size="inline"
+        />
+      </Stack>
+    );
+  }
+
```

Comment:
I tried using a single return here with some conditionals (e.g., `OverlayTrigger` with `show = True` for the `alwaysShowEditButton` version), but it didn't work well.